src/main/java/net/kaczmarzyk/spring/data/jpa/utils/SpecificationBuilder.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2014-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.utils;
+
+import net.kaczmarzyk.spring.data.jpa.web.ProcessingContext;
+import net.kaczmarzyk.spring.data.jpa.web.SpecificationFactory;
+import net.kaczmarzyk.spring.data.jpa.web.StandaloneProcessingContext;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Kacper Leśniak
```

Comment:
consider adding Jakub as well as he was the main advisor

Reply:
add `, Tratif sp. z o.o.` after your name to reflect the fact that the company managed and paid for this feature. Also change in other places

Reply:
also add javadoc -- you can reuse text from changelog, just format it properly for javadoc purpose